executor/src/main/java/com/redhat/service/bridge/executor/Executor.java

```diff
@@ -19,9 +20,12 @@
 
     private final FilterEvaluator filterEvaluator;
 
-    public Executor(ProcessorDTO processor, FilterEvaluatorFactory factory) {
+    private final ActionInvoker actionInvoker;
+
+    public Executor(ProcessorDTO processor, FilterEvaluatorFactory factory, ActionInvoker actionInvoker) {
```

Comment:
I prefer to keep the build method invocation here for both factories: filters, actions and transformations are related to processorDTO so it should not be possible to provide a set of filter or an action that is not defined in the processorDTO while now this is possible because the executor class has no control of actionInvoker